controllers/scheduler_test.go

```diff
@@ -2,291 +2,518 @@ package controllers_test
-				Expect(workplacementList.Items[0].Spec.TargetDestinationName).To(Equal(devDestination.Name))
-				Expect(workplacementList.Items[0].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
-				Expect(workplacementList.Items[0].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
-
-				Expect(workplacementList.Items[1].Name).To(HavePrefix("dev-work-name.dev-2"))
-				Expect(workplacementList.Items[1].Namespace).To(Equal(KratixSystemNamespace))
-				Expect(workplacementList.Items[1].Spec.TargetDestinationName).To(Equal(devDestination2.Name))
-				Expect(workplacementList.Items[1].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
-				Expect(workplacementList.Items[1].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
-
-				Expect(workplacementList.Items[2].Name).To(HavePrefix("dev-work-name.dev3"))
-				Expect(workplacementList.Items[2].Namespace).To(Equal(KratixSystemNamespace))
-				Expect(workplacementList.Items[2].Spec.TargetDestinationName).To(Equal(devDestination3.Name))
-				Expect(workplacementList.Items[2].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
-				Expect(workplacementList.Items[2].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
+				Expect(fakeK8sClient.List(context.Background(), &workPlacements, lo)).To(Succeed())
+
+				// check that the dependency work is scheduled to all destinations, including the new one (dev-3)
+				Expect(workPlacements.Items).To(HaveLen(3))
+
+				Expect(workPlacements.Items[0].Name).To(HavePrefix("dev-work-name.dev-1"))
+				Expect(workPlacements.Items[0].Namespace).To(Equal(KratixSystemNamespace))
+				Expect(workPlacements.Items[0].Spec.TargetDestinationName).To(Equal(devDestination.Name))
+				Expect(workPlacements.Items[0].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
+				Expect(workPlacements.Items[0].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
+
+				Expect(workPlacements.Items[1].Name).To(HavePrefix("dev-work-name.dev-2"))
+				Expect(workPlacements.Items[1].Namespace).To(Equal(KratixSystemNamespace))
+				Expect(workPlacements.Items[1].Spec.TargetDestinationName).To(Equal(devDestination2.Name))
+				Expect(workPlacements.Items[1].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
+				Expect(workPlacements.Items[1].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
+
+				Expect(workPlacements.Items[2].Name).To(HavePrefix("dev-work-name.dev-3"))
+				Expect(workPlacements.Items[2].Namespace).To(Equal(KratixSystemNamespace))
+				Expect(workPlacements.Items[2].Spec.TargetDestinationName).To(Equal(devDestination3.Name))
+				Expect(workPlacements.Items[2].Spec.Workloads).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads))
+				Expect(workPlacements.Items[2].Spec.ID).To(Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID))
```

Comment:
(more fyi than not)

there are multiple tests where we check values on a struct

we could use [gstruct](https://onsi.github.io/gomega/#codegstructcode-testing-complex-data-types) from  Gomega and write something slightly more readable.

For example, see https://github.com/syntasso/kratix/blob/374ba26857874742c8c87ab0e3df110bf2f7a1ed/controllers/promise_and_dynamic_resource_request_controller_test.go#L900-L912

(if we dot import gstruct it can read even better)


This particular snippet may be rewritten as

```go
id := func(index int, _ interface{}) string {
	return strconv.Itoa(index)
}
Expect(workPlacements.Items).To(MatchElementsWithIndex(id, IgnoreExtras, Elements{
	"0": MatchFields(IgnoreExtras, Fields{
		"ObjectMeta": MatchFields(IgnoreExtras, Fields{
			"Name":      HavePrefix("dev-work-name.dev-1"),
			"Namespace": Equal(KratixSystemNamespace),
		}),
		"Spec": MatchFields(IgnoreExtras, Fields{
			"TargetDestinationName": Equal(devDestination.Name),
			"Workloads":             Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].Workloads),
			"ID":                    Equal(dependencyWorkForDev.Spec.WorkloadGroups[0].ID),
		}),
	}),
}))
```

Reply:
Created [186726492](https://www.pivotaltracker.com/story/show/186726492) to start using gstruct ✅